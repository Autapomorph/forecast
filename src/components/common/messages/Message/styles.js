import styled from 'styled-components';

export const StyledMessageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
`;

export const StyledMessageContent = styled.span`
  text-align: center;
  font-size: 1.5rem;
`;
