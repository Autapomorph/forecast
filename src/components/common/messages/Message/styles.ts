import styled from 'styled-components/macro';

export const StyledMessageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
  font-size: 1.3rem;
`;

export const StyledMessageContent = styled.span`
  text-align: center;
`;
