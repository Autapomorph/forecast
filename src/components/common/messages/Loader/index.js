import React from 'react';

import { StyledLoaderMessage } from './styles';

const Loader = ({ children, ...props }) => (
  <StyledLoaderMessage {...props}>{children}</StyledLoaderMessage>
);

Loader.defaultProps = {
  children: 'Загрузка',
};

export default Loader;
