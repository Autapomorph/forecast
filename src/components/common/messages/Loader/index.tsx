import React from 'react';
import MDSpinner from 'react-md-spinner';

import { StyledLoaderMessage } from './styles';

const Loader: React.FC = ({ children, ...props }): React.ReactElement => (
  <StyledLoaderMessage {...props}>{children}</StyledLoaderMessage>
);

Loader.defaultProps = {
  children: <MDSpinner singleColor="#fff" />,
};

export default Loader;
