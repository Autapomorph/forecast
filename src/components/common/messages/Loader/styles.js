import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledLoaderMessage = styled(Message)`
  margin: 10px 0;
`;
