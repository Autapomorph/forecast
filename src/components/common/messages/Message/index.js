import React from 'react';

import { StyledMessageWrapper, StyledMessageContent } from './styles';

const Message = ({ children, ...props }) => (
  <StyledMessageWrapper {...props}>
    <StyledMessageContent>{children}</StyledMessageContent>
  </StyledMessageWrapper>
);

export default Message;
