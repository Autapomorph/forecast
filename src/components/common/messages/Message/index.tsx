import React from 'react';

import { StyledMessageWrapper, StyledMessageContent } from './styles';

const Message: React.FC = ({ children, ...props }): React.ReactElement => (
  <StyledMessageWrapper {...props}>
    <StyledMessageContent>{children}</StyledMessageContent>
  </StyledMessageWrapper>
);

export default Message;
