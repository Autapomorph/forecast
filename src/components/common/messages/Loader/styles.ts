import styled from 'styled-components/macro';

import Message from 'components/common/messages/Message';

export const StyledLoaderMessage = styled(Message)`
  margin: 10px 0;
  font-size: 1.5rem;
`;
