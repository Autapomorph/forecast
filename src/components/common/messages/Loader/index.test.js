import Loader from '.';
import Message from '../Message';

describe('Loader tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<Loader />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render defined content', () => {
    const text = 'text';
    const wrapper = mount(<Loader>{text}</Loader>);

    expect(wrapper.find(Message).text()).toEqual(text);
  });
});
