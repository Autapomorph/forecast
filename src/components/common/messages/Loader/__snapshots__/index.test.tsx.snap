// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader tests renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader>
    <MDSpinner
      color1="rgb(66, 165, 245)"
      color2="rgb(239, 83, 80)"
      color3="rgb(253, 216, 53)"
      color4="rgb(76, 175, 80)"
      duration={1333}
      singleColor="#fff"
      size={28}
    />
  </Loader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <MDSpinner
        color1="rgb(66, 165, 245)"
        color2="rgb(239, 83, 80)"
        color3="rgb(253, 216, 53)"
        color4="rgb(76, 175, 80)"
        duration={1333}
        singleColor="#fff"
        size={28}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color1": "rgb(66, 165, 245)",
        "color2": "rgb(239, 83, 80)",
        "color3": "rgb(253, 216, 53)",
        "color4": "rgb(76, 175, 80)",
        "duration": 1333,
        "singleColor": "#fff",
        "size": 28,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "styles__StyledLoaderMessage-p6qlbu-0",
        "isStatic": true,
        "rules": Array [
          "margin:10px 0;font-size:1.5rem;",
        ],
      },
      "displayName": "styles__StyledLoaderMessage",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "styles__StyledLoaderMessage-p6qlbu-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <MDSpinner
          color1="rgb(66, 165, 245)"
          color2="rgb(239, 83, 80)"
          color3="rgb(253, 216, 53)"
          color4="rgb(76, 175, 80)"
          duration={1333}
          singleColor="#fff"
          size={28}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color1": "rgb(66, 165, 245)",
          "color2": "rgb(239, 83, 80)",
          "color3": "rgb(253, 216, 53)",
          "color4": "rgb(76, 175, 80)",
          "duration": 1333,
          "singleColor": "#fff",
          "size": 28,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "styles__StyledLoaderMessage-p6qlbu-0",
          "isStatic": true,
          "rules": Array [
            "margin:10px 0;font-size:1.5rem;",
          ],
        },
        "displayName": "styles__StyledLoaderMessage",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "styles__StyledLoaderMessage-p6qlbu-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
